python/lib/dependabot/python/file_parser/pyproject_files_parser.rb

```diff
@@ -154,21 +159,49 @@ def lockfile_dependencies
           parsed_lockfile.fetch("package", []).each do |details|
             next if source_types.include?(details.dig("source", "type"))
 
+            name = normalise(details.fetch("name"))
+
             dependencies <<
               Dependency.new(
-                name: normalise(details.fetch("name")),
+                name: name,
                 version: details.fetch("version"),
                 requirements: [],
                 package_manager: "pip",
                 subdependency_metadata: [{
-                  production: details["category"] != "dev"
+                  production: production_dependency_names.include?(name)
                 }]
               )
           end
 
           dependencies
         end
 
+        def production_dependency_names
+          @production_dependency_names ||= parse_production_dependency_names
+        end
+
+        def parse_production_dependency_names
+          SharedHelpers.in_a_temporary_directory do
+            File.write(pyproject.name, pyproject.content)
+            File.write(lockfile.name, lockfile.content)
+
+            begin
+              output = Helpers.run_poetry_command("pyenv exec poetry show --only main")
```

Comment:
👍 for using the poetry CLI rather than inspecting the file directly.

However, the docs for [`show` indicate that `--only main` and `--without dev` are both workable here](https://python-poetry.org/docs/cli/#show)... 

At first I was thinking that `--without dev` might be better here, as `main` is the default group but there may be other groups, but then I saw this note [here](https://python-poetry.org/docs/managing-dependencies/):
> Dependency groups, other than the implicit `main` group, must only contain dependencies you need in your development process. Installing them is only possible by using Poetry.

So I'm guessing docs/tests, etc are other groups that won't be considered part of `main`, but aren't part of `dev` either... 

The :dependabot: concept of dep categories isn't quite so sophisticated, we tend to simply have "prod" vs "dev", and in this method solely want prod deps, not "non-dev deps", so `--only main` makes sense to me!

Reply:
Yeah, in general I tend to consider "dev", everything that's not production/runtime, that's why I went with `--only main`.